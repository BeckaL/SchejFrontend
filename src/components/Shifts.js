import React, { Component } from 'react'
import BigCalendar from 'react-big-calendar'
import NewShiftForm from './NewShiftForm'
import SignOutButton from './SignOutButton'
import moment from 'moment'
import axios from 'axios'
import 'react-big-calendar/lib/css/react-big-calendar.css'
import Popup from "reactjs-popup";
import ShiftPopup from './ShiftPopup'
import NotificationsList from './NotificationsList'
const localizer = BigCalendar.momentLocalizer(moment)

export default class Shifts extends Component {
  constructor(props) {
    super(props)
    this.state = {
      events: [],
      open: false,
      newShiftOpen: false,
      displayedShift: '',
      notificationsOpen: false
    }
    this.openModal = this.openModal.bind(this)
    this.closeModal = this.closeModal.bind(this)
  }

  openModal (shift){
   this.setState({
     open: true,
     displayedShift: {
       shiftId: shift.eventId,
       title: shift.title,
       userId: shift.userId,
       start: shift.start,
       end: shift.end,
       email: shift.email
     }
   })
 }

 openNewShiftModal = () => {
   this.setState({
     newShiftOpen: true
   })
 }
 openNotificationsModal = () => {
   this.setState({
     notificationsOpen: true
   })
 }

 closeModal () {
   this.setState({
     open: false,
     newShiftOpen: false,
     notificationsOpen: false
   })
 }

  componentWillMount() {
    axios.get(
      `${process.env.REACT_APP_API_URL}/shifts?organisation=${localStorage['organisation']}&job_title=${localStorage['jobTitle']}`
    )
    .then(response => {
      console.log(response)
      let shiftData = response.data
      for(var i in shiftData) {
        shiftData[i].start_time = new Date(parseInt(shiftData[i].start_time))
        shiftData[i].end_time = new Date(parseInt(shiftData[i].end_time))

        this.setState(prevState => ({
          events: [...prevState.events, {
            title: shiftData[i].title,
            start: shiftData[i].start_time,
            end: shiftData[i].end_time,
            eventId: shiftData[i].id,
            userId: shiftData[i].user_id,
            email: shiftData[i].email
          }]
        }))
      }
    })
  }

  render() {
    if (!localStorage['authenticationToken']) {
      this.props.history.push('/')
    }
    return(
      <div>
        <h1 id='title'>Schej</h1>
        <section id='welcome'>
          <h2 id='welcome-name'>Welcome {localStorage['name']}</h2>
          <SignOutButton
            history={this.props.history}
          /><br /><br />
          <h3 id='welcome-organisation'>Organisation: {localStorage['organisation']}</h3>

          <button id="add-shift-button" className='custom-button' onClick={this.openNewShiftModal}>Add Shift</button>
          <button id="notifications-button" className='custom-button' onClick={this.openNotificationsModal}>Notifications</button>
        </section>
        <div>
          <BigCalendar
            localizer = { localizer }
            defaultDate = { new Date() }
            selectable
            defaultView = "month"
            onSelectEvent={(shift) => this.openModal(shift)}
            events= { this.state.events }
            style={{ height: '100vh' }}
          />
        </div>
        <Popup
          open={this.state.open}
          closeOnDocumentClick
          onClose={this.closeModal}
        >
          <ShiftPopup
            shiftInfo={this.state.displayedShift}
            history={this.props.history}
          />
        </Popup>
        <Popup
          open={this.state.newShiftOpen}
          closeOnDocumentClick
          onClose={this.closeModal}
        >
          <NewShiftForm
            history={this.props.history}
          />
        </Popup>
        <Popup
          open={this.state.notificationsOpen}
          closeOnDocumentClick
          onClose={this.closeModal}
        >
        <NotificationsList
          history={this.props.history}
          />
        </Popup>
      </div>
  )}
}
