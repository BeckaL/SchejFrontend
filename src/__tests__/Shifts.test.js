import React from 'react'
import { mount, shallow } from 'enzyme'
import BigCalendar from 'react-big-calendar'
import MockDate from 'mockdate'
import Shifts from '../components/Shifts.js'
import axios from 'axios'

jest.mock('axios')

describe('Shifts', () => {
  var wrapper

  beforeAll(() => {
    global.localStorage['authenticationToken'] = 'TestToken'
    const response = {}
    axios.get.mockResolvedValue(response)
    MockDate.set(1545151751163)

    wrapper = shallow(<Shifts history={[]}/>)
  })

  beforeEach(() => {
    global.localStorage.clear()
    global.localStorage['authenticationToken'] = 'TestToken'
    global.localStorage['organisation'] = 'TestOrganisation'
    global.localStorage['jobTitle'] = 'TestTitle'
  })

  test('it matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('it renders the sign out button', () => {
    expect(wrapper.find('SignOutButton').length).toEqual(1)
  })

  test('it renders the notifications button', () => {
    expect(wrapper.find('NotificationsButton').length).toEqual(1)
  })

  test('it renders the BigCalendar plugin component', () => {
    const calendars = wrapper.find(BigCalendar)
    expect(calendars.length).toEqual(1)
  })

  test('it renders the shift popup', () => {
    expect(wrapper.find('ShiftPopup').length).toEqual(1)
  })

  test('it renders the new shift form', () => {
    expect(wrapper.find('NewShiftForm').length).toEqual(1)
  })

  test('it renders 2 popup components', () => {
    const popups = wrapper.find('Popup')
    expect(popups.length).toEqual(2)
  })

  test('it renders the add shift button', () => {
    expect(wrapper.find('#add-shift-button').length).toEqual(1)
  })

  test('it welcomes the user based on the browser local storage', () => {
    global.localStorage['name'] = 'TestName'

    const welcomeName = wrapper.find('#welcome-name')

    setTimeout(() => {
      expect(welcomeName.text()).toEqual('Welcome TestName')
    }, 20)
  })

  test('it shows the user organisation based on browser local storage', () => {
    global.localStorage['organisation'] = 'TestOrganisation'

    const welcomeOrganisation = wrapper.find('#welcome-organisation')

    setTimeout(() => {
      expect(welcomeOrganisation.text()).toEqual(
        'Organisation: TestOrganisation'
      )
    }, 20)
  })

  test('it makes a request to the correct API address', () => {
    const shiftsResponse = {
      data: [
        {
          title: 'TestName',
          email: 'TestEmail',
          organisation: 'TestOrganisation',
          start_time: new Date(),
          end_time: new Date(),
          id: '1',
          user_id: '1'
        }
      ]
    }
    axios.get.mockResolvedValue(shiftsResponse)

    wrapper.instance().getAllShifts()

    expect(axios.get).toHaveBeenCalledTimes(2)
    expect(axios.get).toHaveBeenCalledWith(
      `http://localhost:3001/api/v1/shifts?organisation=${global.localStorage['organisation']}&job_title=${global.localStorage['jobTitle']}`
    )
  })

  test('it makes a request when closing notifications modal', () => {
    const button = wrapper.find('NotificationsButton')
    button.props().onClose()

    setTimeout(() => {
      expect(axios.get).toHaveBeenCalledTimes(3)
      expect(button.props().open).toEqual(false)
    }, 20)
  })

  test('it can open the new shift form by clicking button', () => {
    expect(wrapper.instance().state.newShiftOpen).toEqual(false)

    const button = wrapper.find('#add-shift-button')
    button.simulate('click')

    expect(wrapper.instance().state.newShiftOpen).toEqual(true)
  })

  test('it can close the new shift form', () => {
    const button = wrapper.find('#add-shift-button')
    button.simulate('click')

    wrapper.instance().closeModal()

    expect(wrapper.instance().state.newShiftOpen).toEqual(false)
  })

  test('it can open the correct shift on calendar when clicked', () => {
    const shift = {
      eventId: 1,
      title: 'TestTitle',
      userId: 1,
      start: new Date(),
      end: new Date(),
      email: 'TestEmail'
    }
    expect(wrapper.instance().state.events).toEqual([])

    wrapper.find(BigCalendar).props().onSelectEvent(shift)

    setTimeout(() => {
      expect(wrapper.instance().state.events).toEqual([shift])
    }, 10)
  })

  afterAll(() => {
    MockDate.reset()
    localStorage.clear()
  })
})
