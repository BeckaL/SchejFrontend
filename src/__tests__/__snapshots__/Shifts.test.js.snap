// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shifts it matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Shifts
    history={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          id="title"
          style={
            Object {
              "top": "50px",
            }
          }
        >
          Schej
        </h1>,
        <section
          id="welcome"
        >
          <h2
            id="welcome-name"
          >
            Welcome 
            <span
              id="user-name-on-hompage"
            />
          </h2>
          <SignOutButton
            history={Array []}
          />
          <br />
          <br />
          <h3
            id="welcome-organisation"
          >
            Organisation: 
          </h3>
          <button
            className="custom-button"
            id="add-shift-button"
            onClick={[Function]}
          >
            Add Shift
          </button>
          <NotificationsButton
            onClose={[Function]}
          />
        </section>,
        <div>
          <ForwardRef
            defaultDate={2018-12-18T16:49:11.163Z}
            defaultView="month"
            localizer={
              DateLocalizer {
                "format": [Function],
                "formats": Object {
                  "agendaDateFormat": "ddd MMM DD",
                  "agendaHeaderFormat": [Function],
                  "agendaTimeFormat": "LT",
                  "agendaTimeRangeFormat": [Function],
                  "dateFormat": "DD",
                  "dayFormat": "DD ddd",
                  "dayHeaderFormat": "dddd MMM DD",
                  "dayRangeHeaderFormat": [Function],
                  "eventTimeRangeEndFormat": [Function],
                  "eventTimeRangeFormat": [Function],
                  "eventTimeRangeStartFormat": [Function],
                  "monthHeaderFormat": "MMMM YYYY",
                  "selectRangeFormat": [Function],
                  "timeGutterFormat": "LT",
                  "weekdayFormat": "ddd",
                },
                "propType": [Function],
                "startOfWeek": [Function],
              }
            }
            onSelectEvent={[Function]}
            selectable={true}
            style={
              Object {
                "height": "77vh",
              }
            }
          />
        </div>,
        <Popup
          arrow={true}
          arrowStyle={Object {}}
          className=""
          closeOnDocumentClick={true}
          closeOnEscape={true}
          contentStyle={Object {}}
          defaultOpen={false}
          disabled={false}
          keepTooltipInside={false}
          lockScroll={false}
          modal={false}
          mouseEnterDelay={100}
          mouseLeaveDelay={100}
          offsetX={0}
          offsetY={0}
          on={
            Array [
              "click",
            ]
          }
          onClose={[Function]}
          onOpen={[Function]}
          open={false}
          overlayStyle={Object {}}
          position="bottom center"
          repositionOnResize={true}
          trigger={null}
        >
          <ShiftPopup
            history={Array []}
            shiftInfo=""
          />
        </Popup>,
        <Popup
          arrow={true}
          arrowStyle={Object {}}
          className=""
          closeOnDocumentClick={true}
          closeOnEscape={true}
          contentStyle={Object {}}
          defaultOpen={false}
          disabled={false}
          keepTooltipInside={false}
          lockScroll={false}
          modal={false}
          mouseEnterDelay={100}
          mouseLeaveDelay={100}
          offsetX={0}
          offsetY={0}
          on={
            Array [
              "click",
            ]
          }
          onClose={[Function]}
          onOpen={[Function]}
          open={false}
          overlayStyle={Object {}}
          position="bottom center"
          repositionOnResize={true}
          trigger={null}
        >
          <NewShiftForm
            history={Array []}
          />
        </Popup>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Schej",
          "id": "title",
          "style": Object {
            "top": "50px",
          },
        },
        "ref": null,
        "rendered": "Schej",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              id="welcome-name"
            >
              Welcome 
              <span
                id="user-name-on-hompage"
              />
            </h2>,
            <SignOutButton
              history={Array []}
            />,
            <br />,
            <br />,
            <h3
              id="welcome-organisation"
            >
              Organisation: 
            </h3>,
            <button
              className="custom-button"
              id="add-shift-button"
              onClick={[Function]}
            >
              Add Shift
            </button>,
            <NotificationsButton
              onClose={[Function]}
            />,
          ],
          "id": "welcome",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Welcome ",
                <span
                  id="user-name-on-hompage"
                />,
              ],
              "id": "welcome-name",
            },
            "ref": null,
            "rendered": Array [
              "Welcome ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "id": "user-name-on-hompage",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "history": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Organisation: ",
                undefined,
              ],
              "id": "welcome-organisation",
            },
            "ref": null,
            "rendered": Array [
              "Organisation: ",
              undefined,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add Shift",
              "className": "custom-button",
              "id": "add-shift-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add Shift",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClose": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ForwardRef
            defaultDate={2018-12-18T16:49:11.163Z}
            defaultView="month"
            localizer={
              DateLocalizer {
                "format": [Function],
                "formats": Object {
                  "agendaDateFormat": "ddd MMM DD",
                  "agendaHeaderFormat": [Function],
                  "agendaTimeFormat": "LT",
                  "agendaTimeRangeFormat": [Function],
                  "dateFormat": "DD",
                  "dayFormat": "DD ddd",
                  "dayHeaderFormat": "dddd MMM DD",
                  "dayRangeHeaderFormat": [Function],
                  "eventTimeRangeEndFormat": [Function],
                  "eventTimeRangeFormat": [Function],
                  "eventTimeRangeStartFormat": [Function],
                  "monthHeaderFormat": "MMMM YYYY",
                  "selectRangeFormat": [Function],
                  "timeGutterFormat": "LT",
                  "weekdayFormat": "ddd",
                },
                "propType": [Function],
                "startOfWeek": [Function],
              }
            }
            onSelectEvent={[Function]}
            selectable={true}
            style={
              Object {
                "height": "77vh",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "defaultDate": 2018-12-18T16:49:11.163Z,
            "defaultView": "month",
            "events": undefined,
            "localizer": DateLocalizer {
              "format": [Function],
              "formats": Object {
                "agendaDateFormat": "ddd MMM DD",
                "agendaHeaderFormat": [Function],
                "agendaTimeFormat": "LT",
                "agendaTimeRangeFormat": [Function],
                "dateFormat": "DD",
                "dayFormat": "DD ddd",
                "dayHeaderFormat": "dddd MMM DD",
                "dayRangeHeaderFormat": [Function],
                "eventTimeRangeEndFormat": [Function],
                "eventTimeRangeFormat": [Function],
                "eventTimeRangeStartFormat": [Function],
                "monthHeaderFormat": "MMMM YYYY",
                "selectRangeFormat": [Function],
                "timeGutterFormat": "LT",
                "weekdayFormat": "ddd",
              },
              "propType": [Function],
              "startOfWeek": [Function],
            },
            "onSelectEvent": [Function],
            "selectable": true,
            "style": Object {
              "height": "77vh",
            },
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "ControlledComponent": [Function],
            "Navigate": Object {
              "DATE": "DATE",
              "NEXT": "NEXT",
              "PREVIOUS": "PREV",
              "TODAY": "TODAY",
            },
            "Views": Object {
              "AGENDA": "agenda",
              "DAY": "day",
              "MONTH": "month",
              "WEEK": "week",
              "WORK_WEEK": "work_week",
            },
            "components": Object {
              "dateCellWrapper": [Function],
              "dayWrapper": [Function],
              "eventWrapper": [Function],
            },
            "deferControlTo": [Function],
            "globalizeLocalizer": [Function],
            "momentLocalizer": [Function],
            "move": [Function],
            "propTypes": Object {
              "date": [Function],
              "defaultDate": [Function],
              "defaultSelected": [Function],
              "defaultView": [Function],
              "innerRef": [Function],
              "selected": [Function],
              "view": [Function],
            },
            "render": [Function],
          },
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrow": true,
          "arrowStyle": Object {},
          "children": <ShiftPopup
            history={Array []}
            shiftInfo=""
          />,
          "className": "",
          "closeOnDocumentClick": true,
          "closeOnEscape": true,
          "contentStyle": Object {},
          "defaultOpen": false,
          "disabled": false,
          "keepTooltipInside": false,
          "lockScroll": false,
          "modal": false,
          "mouseEnterDelay": 100,
          "mouseLeaveDelay": 100,
          "offsetX": 0,
          "offsetY": 0,
          "on": Array [
            "click",
          ],
          "onClose": [Function],
          "onOpen": [Function],
          "open": false,
          "overlayStyle": Object {},
          "position": "bottom center",
          "repositionOnResize": true,
          "trigger": null,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "history": Array [],
            "shiftInfo": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrow": true,
          "arrowStyle": Object {},
          "children": <NewShiftForm
            history={Array []}
          />,
          "className": "",
          "closeOnDocumentClick": true,
          "closeOnEscape": true,
          "contentStyle": Object {},
          "defaultOpen": false,
          "disabled": false,
          "keepTooltipInside": false,
          "lockScroll": false,
          "modal": false,
          "mouseEnterDelay": 100,
          "mouseLeaveDelay": 100,
          "offsetX": 0,
          "offsetY": 0,
          "on": Array [
            "click",
          ],
          "onClose": [Function],
          "onOpen": [Function],
          "open": false,
          "overlayStyle": Object {},
          "position": "bottom center",
          "repositionOnResize": true,
          "trigger": null,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "history": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            id="title"
            style={
              Object {
                "top": "50px",
              }
            }
          >
            Schej
          </h1>,
          <section
            id="welcome"
          >
            <h2
              id="welcome-name"
            >
              Welcome 
              <span
                id="user-name-on-hompage"
              />
            </h2>
            <SignOutButton
              history={Array []}
            />
            <br />
            <br />
            <h3
              id="welcome-organisation"
            >
              Organisation: 
            </h3>
            <button
              className="custom-button"
              id="add-shift-button"
              onClick={[Function]}
            >
              Add Shift
            </button>
            <NotificationsButton
              onClose={[Function]}
            />
          </section>,
          <div>
            <ForwardRef
              defaultDate={2018-12-18T16:49:11.163Z}
              defaultView="month"
              localizer={
                DateLocalizer {
                  "format": [Function],
                  "formats": Object {
                    "agendaDateFormat": "ddd MMM DD",
                    "agendaHeaderFormat": [Function],
                    "agendaTimeFormat": "LT",
                    "agendaTimeRangeFormat": [Function],
                    "dateFormat": "DD",
                    "dayFormat": "DD ddd",
                    "dayHeaderFormat": "dddd MMM DD",
                    "dayRangeHeaderFormat": [Function],
                    "eventTimeRangeEndFormat": [Function],
                    "eventTimeRangeFormat": [Function],
                    "eventTimeRangeStartFormat": [Function],
                    "monthHeaderFormat": "MMMM YYYY",
                    "selectRangeFormat": [Function],
                    "timeGutterFormat": "LT",
                    "weekdayFormat": "ddd",
                  },
                  "propType": [Function],
                  "startOfWeek": [Function],
                }
              }
              onSelectEvent={[Function]}
              selectable={true}
              style={
                Object {
                  "height": "77vh",
                }
              }
            />
          </div>,
          <Popup
            arrow={true}
            arrowStyle={Object {}}
            className=""
            closeOnDocumentClick={true}
            closeOnEscape={true}
            contentStyle={Object {}}
            defaultOpen={false}
            disabled={false}
            keepTooltipInside={false}
            lockScroll={false}
            modal={false}
            mouseEnterDelay={100}
            mouseLeaveDelay={100}
            offsetX={0}
            offsetY={0}
            on={
              Array [
                "click",
              ]
            }
            onClose={[Function]}
            onOpen={[Function]}
            open={false}
            overlayStyle={Object {}}
            position="bottom center"
            repositionOnResize={true}
            trigger={null}
          >
            <ShiftPopup
              history={Array []}
              shiftInfo=""
            />
          </Popup>,
          <Popup
            arrow={true}
            arrowStyle={Object {}}
            className=""
            closeOnDocumentClick={true}
            closeOnEscape={true}
            contentStyle={Object {}}
            defaultOpen={false}
            disabled={false}
            keepTooltipInside={false}
            lockScroll={false}
            modal={false}
            mouseEnterDelay={100}
            mouseLeaveDelay={100}
            offsetX={0}
            offsetY={0}
            on={
              Array [
                "click",
              ]
            }
            onClose={[Function]}
            onOpen={[Function]}
            open={false}
            overlayStyle={Object {}}
            position="bottom center"
            repositionOnResize={true}
            trigger={null}
          >
            <NewShiftForm
              history={Array []}
            />
          </Popup>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Schej",
            "id": "title",
            "style": Object {
              "top": "50px",
            },
          },
          "ref": null,
          "rendered": "Schej",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                id="welcome-name"
              >
                Welcome 
                <span
                  id="user-name-on-hompage"
                />
              </h2>,
              <SignOutButton
                history={Array []}
              />,
              <br />,
              <br />,
              <h3
                id="welcome-organisation"
              >
                Organisation: 
              </h3>,
              <button
                className="custom-button"
                id="add-shift-button"
                onClick={[Function]}
              >
                Add Shift
              </button>,
              <NotificationsButton
                onClose={[Function]}
              />,
            ],
            "id": "welcome",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Welcome ",
                  <span
                    id="user-name-on-hompage"
                  />,
                ],
                "id": "welcome-name",
              },
              "ref": null,
              "rendered": Array [
                "Welcome ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "id": "user-name-on-hompage",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "history": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Organisation: ",
                  undefined,
                ],
                "id": "welcome-organisation",
              },
              "ref": null,
              "rendered": Array [
                "Organisation: ",
                undefined,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Shift",
                "className": "custom-button",
                "id": "add-shift-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Add Shift",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClose": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ForwardRef
              defaultDate={2018-12-18T16:49:11.163Z}
              defaultView="month"
              localizer={
                DateLocalizer {
                  "format": [Function],
                  "formats": Object {
                    "agendaDateFormat": "ddd MMM DD",
                    "agendaHeaderFormat": [Function],
                    "agendaTimeFormat": "LT",
                    "agendaTimeRangeFormat": [Function],
                    "dateFormat": "DD",
                    "dayFormat": "DD ddd",
                    "dayHeaderFormat": "dddd MMM DD",
                    "dayRangeHeaderFormat": [Function],
                    "eventTimeRangeEndFormat": [Function],
                    "eventTimeRangeFormat": [Function],
                    "eventTimeRangeStartFormat": [Function],
                    "monthHeaderFormat": "MMMM YYYY",
                    "selectRangeFormat": [Function],
                    "timeGutterFormat": "LT",
                    "weekdayFormat": "ddd",
                  },
                  "propType": [Function],
                  "startOfWeek": [Function],
                }
              }
              onSelectEvent={[Function]}
              selectable={true}
              style={
                Object {
                  "height": "77vh",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "defaultDate": 2018-12-18T16:49:11.163Z,
              "defaultView": "month",
              "events": undefined,
              "localizer": DateLocalizer {
                "format": [Function],
                "formats": Object {
                  "agendaDateFormat": "ddd MMM DD",
                  "agendaHeaderFormat": [Function],
                  "agendaTimeFormat": "LT",
                  "agendaTimeRangeFormat": [Function],
                  "dateFormat": "DD",
                  "dayFormat": "DD ddd",
                  "dayHeaderFormat": "dddd MMM DD",
                  "dayRangeHeaderFormat": [Function],
                  "eventTimeRangeEndFormat": [Function],
                  "eventTimeRangeFormat": [Function],
                  "eventTimeRangeStartFormat": [Function],
                  "monthHeaderFormat": "MMMM YYYY",
                  "selectRangeFormat": [Function],
                  "timeGutterFormat": "LT",
                  "weekdayFormat": "ddd",
                },
                "propType": [Function],
                "startOfWeek": [Function],
              },
              "onSelectEvent": [Function],
              "selectable": true,
              "style": Object {
                "height": "77vh",
              },
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "ControlledComponent": [Function],
              "Navigate": Object {
                "DATE": "DATE",
                "NEXT": "NEXT",
                "PREVIOUS": "PREV",
                "TODAY": "TODAY",
              },
              "Views": Object {
                "AGENDA": "agenda",
                "DAY": "day",
                "MONTH": "month",
                "WEEK": "week",
                "WORK_WEEK": "work_week",
              },
              "components": Object {
                "dateCellWrapper": [Function],
                "dayWrapper": [Function],
                "eventWrapper": [Function],
              },
              "deferControlTo": [Function],
              "globalizeLocalizer": [Function],
              "momentLocalizer": [Function],
              "move": [Function],
              "propTypes": Object {
                "date": [Function],
                "defaultDate": [Function],
                "defaultSelected": [Function],
                "defaultView": [Function],
                "innerRef": [Function],
                "selected": [Function],
                "view": [Function],
              },
              "render": [Function],
            },
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "arrow": true,
            "arrowStyle": Object {},
            "children": <ShiftPopup
              history={Array []}
              shiftInfo=""
            />,
            "className": "",
            "closeOnDocumentClick": true,
            "closeOnEscape": true,
            "contentStyle": Object {},
            "defaultOpen": false,
            "disabled": false,
            "keepTooltipInside": false,
            "lockScroll": false,
            "modal": false,
            "mouseEnterDelay": 100,
            "mouseLeaveDelay": 100,
            "offsetX": 0,
            "offsetY": 0,
            "on": Array [
              "click",
            ],
            "onClose": [Function],
            "onOpen": [Function],
            "open": false,
            "overlayStyle": Object {},
            "position": "bottom center",
            "repositionOnResize": true,
            "trigger": null,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "history": Array [],
              "shiftInfo": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "arrow": true,
            "arrowStyle": Object {},
            "children": <NewShiftForm
              history={Array []}
            />,
            "className": "",
            "closeOnDocumentClick": true,
            "closeOnEscape": true,
            "contentStyle": Object {},
            "defaultOpen": false,
            "disabled": false,
            "keepTooltipInside": false,
            "lockScroll": false,
            "modal": false,
            "mouseEnterDelay": 100,
            "mouseLeaveDelay": 100,
            "offsetX": 0,
            "offsetY": 0,
            "on": Array [
              "click",
            ],
            "onClose": [Function],
            "onOpen": [Function],
            "open": false,
            "overlayStyle": Object {},
            "position": "bottom center",
            "repositionOnResize": true,
            "trigger": null,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "history": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
